src/EnergyPlus/ConvectionCoefficients.cc

```diff
@@ -109,105 +112,105 @@ namespace ConvectionCoefficients {
-    Real64 const AdaptiveHcOutsideLowLimit(1.0); // W/m2-K
+    Real64 constexpr AdaptiveHcInsideLowLimit{0.5};  // W/m2-K
+    Real64 constexpr AdaptiveHcOutsideLowLimit{1.0}; // W/m2-K
 
     static std::string const BlankString;
 
-    Real64 const OneThird(1.0 / 3.0);   // 1/3 in highest precision
-    Real64 const OneFourth(1.0 / 4.0);  // 1/4 in highest precision
-    Real64 const OneFifth(1.0 / 5.0);   // 1/5 in highest precision
-    Real64 const OneSixth(1.0 / 6.0);   // 1/6 in highest precision
-    Real64 const FourFifths(4.0 / 5.0); // 4/5 in highest precision
+    Real64 constexpr OneThird{1.0 / 3.0};   // 1/3 in highest precision
+    Real64 constexpr OneFourth{1.0 / 4.0};  // 1/4 in highest precision
+    Real64 constexpr OneFifth{1.0 / 5.0};   // 1/5 in highest precision
+    Real64 constexpr OneSixth{1.0 / 6.0};   // 1/6 in highest precision
+    Real64 constexpr FourFifths{4.0 / 5.0}; // 4/5 in highest precision
 
     // Coefficients that modify the convection coeff based on surface roughness
     Array1D<Real64> const RoughnessMultiplier(6, {2.17, 1.67, 1.52, 1.13, 1.11, 1.0});
 
     // parameters for identifying more specific hc model equations, inside face
-    int const HcInt_UserValue(200);
-    int const HcInt_UserSchedule(201);
-    int const HcInt_UserCurve(202);
-    int const HcInt_ASHRAEVerticalWall(203);
-    int const HcInt_WaltonUnstableHorizontalOrTilt(204);
-    int const HcInt_WaltonStableHorizontalOrTilt(205);
-    int const HcInt_FisherPedersenCeilDiffuserFloor(206);
-    int const HcInt_FisherPedersenCeilDiffuserCeiling(207);
-    int const HcInt_FisherPedersenCeilDiffuserWalls(208);
-    int const HcInt_AlamdariHammondStableHorizontal(209);
-    int const HcInt_AlamdariHammondVerticalWall(210);
-    int const HcInt_AlamdariHammondUnstableHorizontal(211);
-    int const HcInt_KhalifaEq3WallAwayFromHeat(212);
-    int const HcInt_KhalifaEq4CeilingAwayFromHeat(213);
-    int const HcInt_KhalifaEq5WallNearHeat(214);
-    int const HcInt_KhalifaEq6NonHeatedWalls(215);
-    int const HcInt_KhalifaEq7Ceiling(216);
-    int const HcInt_AwbiHattonHeatedFloor(217);
-    int const HcInt_AwbiHattonHeatedWall(218);
-    int const HcInt_BeausoleilMorrisonMixedAssistingWall(219);
-    int const HcInt_BeausoleilMorrisonMixedOppossingWall(220);
-    int const HcInt_BeausoleilMorrisonMixedStableCeiling(221);
-    int const HcInt_BeausoleilMorrisonMixedUnstableCeiling(222);
-    int const HcInt_BeausoleilMorrisonMixedStableFloor(223);
-    int const HcInt_BeausoleilMorrisonMixedUnstableFloor(224);
-    int const HcInt_FohannoPolidoriVerticalWall(225);
-    int const HcInt_KaradagChilledCeiling(226);
-    int const HcInt_ISO15099Windows(227);
-    int const HcInt_GoldsteinNovoselacCeilingDiffuserWindow(228);
-    int const HcInt_GoldsteinNovoselacCeilingDiffuserWalls(229);
-    int const HcInt_GoldsteinNovoselacCeilingDiffuserFloor(230);
+    int constexpr HcInt_UserValue{200};
```

Comment:
Could these be moved to enum classes?

Reply:
We did try moving these to an enum class. The problem arises with `VALUE`, and `SCHEDULE` which EnergyPlus expects both to be a value of -999. We cannot have a map with strings mapping to different types (a mix of enums and ints) and we cants make two enums correspond to the same integer. There is definitely some room to refactor this but it will take a more thorough pass through the code and we felt that trying to refactor that far was really getting beyond the scope of what we were trying to accomplish here.

Reply:
OK, that's fine. There's already plenty done here, so we can tackle that issue later.